src/Microsoft.PowerShell.Commands.Management/commands/management/Computer.cs

```diff
@@ -133,6 +133,177 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
         #endregion Serialization
     }
 
+    /// <summary>
+    /// Supported reasons for restarting or shutting down.
+    /// </summary>
+    public enum ShutdownReasons: uint
+    {
+        /// <summary>
+        /// Planned activity.
+        /// </summary>
+        Planned = 0x80000000,
+
+        /// <summary>
+        /// Any other activity.
+        /// </summary>
+        Other = 0x00000000,
+
+        /// <summary>
+        /// Application related shutdown/restart.
+        /// </summary>
+        Application = 0x00040000,
+
+        /// <summary>
+        /// Hardware related shutdown/restart.
+        /// </summary>
+        Hardware = 0x80010000,
+
+        /// <summary>
+        /// OperatingSystem related shutdown/restart.
+        /// </summary>        
+        OperatingSystem = 0x80020000,
+
+        /// <summary>
+        /// Power related shutdown/restart.
+        /// </summary>        
+        Power = 0x80060000,
+
+        /// <summary>
+        /// Software related shutdown/restart.
+        /// </summary>        
+        Software = 0x80030000,
+
+        /// <summary>
+        /// System related shutdown/restart.
+        /// </summary>        
+        System = 0x80050000,
```

Comment:
I have doubts that we should allow all these values. Perhaps the best options is the ones offered by WIndows GUI.

Reply:
was referring to previous discussions in the old PR
https://github.com/PowerShell/PowerShell/pull/8112#discussion_r236122635

Thought its an agreement to go ahead with the entire list.

Reply:
Ok deal